File: [프로그래머스] 스택큐_다리를 지나는 트럭.py
from collections import deque
def solution(bridge_length, weight, truck_weights):
    answer = 0
    truck_weights = deque(truck_weights)
    ing = deque(0 for i in range(bridge_length))
    check = 0
    while ing:
        answer += 1
        check -= ing.popleft()
        if truck_weights: 
            if check + truck_weights[0] <= weight:
                ing.append(truck_weights.popleft())
                check += ing[-1]
            else:
                ing.append(0)
    return answer

File: [프로그래머스] 스택큐_다리를 지나는 트럭.js
function solution(bridge_length, weight, truck_weights) {
    var answer = 0;
    let ing = Array(bridge_length).fill(0);
    let check = 0;
    while(ing.length > 0){
        answer++;
        check -= ing.shift();
        if(truck_weights.length > 0){
            if(check + truck_weights[0] <= weight){
                ing.push(truck_weights.shift());
                check += ing[ing.length - 1];
            } else {
                ing.push(0);
            }
        }
    }
    return answer;
}